src/enc.rs

```diff
@@ -32,6 +32,20 @@ impl Enc {
         let dec = key_pair.backwards * self.inner * key_pair.forwards;
         dec[0].w.0
     }
+
+    #[inline]
+    fn enc_i32(key_pair: &KeyPair, value: i32) -> Self {
```

Comment:
Hmm, it looks like an ideal candidate for a generic. Could we perhaps make `enc` generic, thus accepting either `i32` or `u32`? In the future, we could think of some constraints, but for now, a simple generic type would work OK IMHO.

Reply:
Another thing, are `enc_i32` and `dec_i32` used anywhere? They don't seem though :thinking: